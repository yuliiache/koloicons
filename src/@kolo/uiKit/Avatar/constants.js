export const AVATAR_SIZES = Object.freeze({
  WIDTH: 160,
  HEIGHT: 160,
});
