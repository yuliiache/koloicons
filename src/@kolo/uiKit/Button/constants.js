export const BUTTON_TEST_ID = 'buttonComponent';
export const BUTTON_TEXT_TEST_ID = `${BUTTON_TEST_ID}Text`;

export const BUTTON_COLOR = Object.freeze({
  PRIMARY: 'primary',
  ACCENT: 'accent',
  BASIC: 'basic',
});

export const BUTTON_SIZE = Object.freeze({
  SMALL: 'small',
  BIG: 'big',
  FULL_WIDTH: 'full-width',
});

export const BUTTON_TYPE = Object.freeze({
  BUTTON: 'button',
  SUBMIT: 'submit',
  RESET: 'reset',
});
