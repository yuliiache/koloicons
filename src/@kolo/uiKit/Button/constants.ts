export const BUTTON_TEST_ID = 'buttonComponent';
export const BUTTON_TEXT_TEST_ID = `${BUTTON_TEST_ID}Text`;

export enum ButtonColor {
  PRIMARY = 'primary',
  ACCENT = 'accent',
  BASIC = 'basic',
}

export enum ButtonSize {
  SMALL = 'small',
  BIG = 'big',
  FULL_WIDTH = 'full-width',
}

export enum ButtonType {
  BUTTON = 'button',
  SUBMIT = 'submit',
  RESET = 'reset',
}
