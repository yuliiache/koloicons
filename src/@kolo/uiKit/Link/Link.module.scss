@import "@kolo/design";

.link {
  display: inline-block;

  @include main-text(0, 1.5, null);
  color: $lighthGrey;

  transition: color $transition-main ease-in-out;

  &:hover {
    color: $BlueColor;
  }

  &:focus {
    outline: 0;
    box-shadow: $LinkBoxShadow;
  }

  &--main {
    @include main-text(0.8rem, null, 500);
    color: $secFontColor;
  }

  &--brand {
    @include main-text(0.7rem, null, null);
    color: $BlueColor;

    &:hover {
      color: $GreenColor;
    }
  }

  &--default {
    @include main-text(0.6rem, 1.5, null);
  }

  &--accent {
    @include main-text(0.6rem, 1.5, null);
    color: $GreenColor;
  }

  &--default-red {
    @include main-text(0.6rem, 1.5, null);
    color: $secFontColor;

    &:hover {
      color: $RedColor;
    }
  }
}
