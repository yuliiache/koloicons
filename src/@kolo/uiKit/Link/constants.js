export const INTERNAL_LINK_TEST_ID = 'internalLinkComponent';

export const EXTERNAL_LINK_TEST_ID = 'externalLinkComponent';

export const LINK_TYPES = Object.freeze({
  MAIN: 'main',
  BRAND: 'brand',
  DEFAULT: 'default',
  ACCENT: 'accent',
  DEFAULT_RED: 'default-red',
});
