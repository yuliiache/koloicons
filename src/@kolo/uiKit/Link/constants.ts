export const INTERNAL_LINK_TEST_ID = 'internalLinkComponent';

export const EXTERNAL_LINK_TEST_ID = 'externalLinkComponent';

export enum LinkType {
  MAIN = 'main',
  BRAND = 'brand',
  DEFAULT = 'default',
  ACCENT = 'accent',
  DEFAULT_RED = 'default-red',
}
