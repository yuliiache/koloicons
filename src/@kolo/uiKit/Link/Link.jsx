import React from 'react';
import {Link as InternalLink} from 'react-router-dom';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import {EXTERNAL_LINK_TEST_ID, INTERNAL_LINK_TEST_ID, LINK_TYPES} from './constants';

import styles from './Link.module.scss';

const Link = ({url, children, type, isInNewTab = false}) => {
  const isInternalUrl = url.startsWith('/');
  const linksClasses = classNames(styles.link, styles[`link--${type}`]);

  if (isInternalUrl) {
    return (
      <InternalLink
        className={linksClasses}
        data-testid={INTERNAL_LINK_TEST_ID}
        to={url}
        target={isInNewTab ? '_blank' : undefined}
        rel={isInNewTab ? 'noopener noreferrer' : undefined}
      >
        {children}
      </InternalLink>
    );
  }

  return (
    <a
      className={linksClasses}
      href={url}
      data-testid={EXTERNAL_LINK_TEST_ID}
      target={isInNewTab ? '_blank' : undefined}
      rel="noopener noreferrer nofollow"
    >
      {children}
    </a>
  );
};

Link.propTypes = {
  url: PropTypes.string.isRequired,
  isInNewTab: PropTypes.bool,
  children: PropTypes.node.isRequired,
  type: PropTypes.oneOf(Object.values(LINK_TYPES)),
};

export default Link;
