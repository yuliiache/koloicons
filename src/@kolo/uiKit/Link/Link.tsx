import classNames from 'classnames';
import {FC} from 'react';
import {Link as InternalLink} from 'react-router-dom';

import {EXTERNAL_LINK_TEST_ID, INTERNAL_LINK_TEST_ID, LinkType} from './constants';
import styles from './Link.module.scss';

interface Props {
  url: string;
  children: React.ReactNode;
  type?: LinkType;
  isInNewTab?: boolean;
}

const Link: FC<Props> = ({url, children, type = LinkType.DEFAULT, isInNewTab = false}) => {
  const isInternalUrl: boolean = url.startsWith('/');
  const linksClasses = classNames(styles.link, styles[`link--${type}`]);

  if (isInternalUrl) {
    return (
      <InternalLink
        className={linksClasses}
        data-testid={INTERNAL_LINK_TEST_ID}
        to={url}
        target={isInNewTab ? '_blank' : undefined}
        rel={isInNewTab ? 'noopener noreferrer' : undefined}
      >
        {children}
      </InternalLink>
    );
  }

  return (
    <a
      className={linksClasses}
      href={url}
      data-testid={EXTERNAL_LINK_TEST_ID}
      target={isInNewTab ? '_blank' : undefined}
      rel="noopener noreferrer nofollow"
    >
      {children}
    </a>
  );
};

export default Link;
