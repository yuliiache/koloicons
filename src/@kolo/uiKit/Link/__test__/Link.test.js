import React from 'react';
import {BrowserRouter} from 'react-router-dom';
import {fireEvent, render, screen} from '@testing-library/react';

import {LinkType} from '../constants';
import Link from '../Link';

describe('Link component', () => {
  const renderLink = (props) => render(<Link {...props} />, {wrapper: BrowserRouter});

  Object.values(LinkType).forEach((type) => {
    ['internal', 'external'].forEach((linkType) => {
      const url = linkType === 'internal' ? '/' : 'http://external.com';
      const text = linkType === 'internal' ? 'Internal link' : 'External link';
      const props = {
        url,
        type,
        children: text,
      };

      it(`renders correctly for ${linkType} link with type ${type}`, () => {
        const {asFragment} = renderLink(props);

        expect(asFragment()).toMatchSnapshot();
      });

      it(`renders correctly for ${linkType} link with type ${type} that should open in a new tab`, () => {
        const updatedProps = {
          ...props,
          isInNewTab: true,
        };

        const {asFragment} = renderLink(updatedProps);

        expect(asFragment()).toMatchSnapshot();
      });

      it('should call onClick function when user click the link', () => {
        const onClickMock = jest.fn();
        const updatedProps = {
          ...props,
          onClick: onClickMock,
        };
        renderLink(updatedProps);

        const link = screen.getByText(text);

        fireEvent.click(link);

        expect(onClickMock).toHaveBeenCalled();
      });
    });
  });
});
