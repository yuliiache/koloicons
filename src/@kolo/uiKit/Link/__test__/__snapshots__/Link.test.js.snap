// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component renders correctly for external link with type accent 1`] = `
<DocumentFragment>
  <a
    class="link link--accent"
    data-testid="externalLinkComponent"
    href="http://external.com"
    rel="noopener noreferrer nofollow"
  >
    External link
  </a>
</DocumentFragment>
`;

exports[`Link component renders correctly for external link with type accent that should open in a new tab 1`] = `
<DocumentFragment>
  <a
    class="link link--accent"
    data-testid="externalLinkComponent"
    href="http://external.com"
    rel="noopener noreferrer nofollow"
    target="_blank"
  >
    External link
  </a>
</DocumentFragment>
`;

exports[`Link component renders correctly for external link with type brand 1`] = `
<DocumentFragment>
  <a
    class="link link--brand"
    data-testid="externalLinkComponent"
    href="http://external.com"
    rel="noopener noreferrer nofollow"
  >
    External link
  </a>
</DocumentFragment>
`;

exports[`Link component renders correctly for external link with type brand that should open in a new tab 1`] = `
<DocumentFragment>
  <a
    class="link link--brand"
    data-testid="externalLinkComponent"
    href="http://external.com"
    rel="noopener noreferrer nofollow"
    target="_blank"
  >
    External link
  </a>
</DocumentFragment>
`;

exports[`Link component renders correctly for external link with type default 1`] = `
<DocumentFragment>
  <a
    class="link link--default"
    data-testid="externalLinkComponent"
    href="http://external.com"
    rel="noopener noreferrer nofollow"
  >
    External link
  </a>
</DocumentFragment>
`;

exports[`Link component renders correctly for external link with type default that should open in a new tab 1`] = `
<DocumentFragment>
  <a
    class="link link--default"
    data-testid="externalLinkComponent"
    href="http://external.com"
    rel="noopener noreferrer nofollow"
    target="_blank"
  >
    External link
  </a>
</DocumentFragment>
`;

exports[`Link component renders correctly for external link with type default-red 1`] = `
<DocumentFragment>
  <a
    class="link link--default-red"
    data-testid="externalLinkComponent"
    href="http://external.com"
    rel="noopener noreferrer nofollow"
  >
    External link
  </a>
</DocumentFragment>
`;

exports[`Link component renders correctly for external link with type default-red that should open in a new tab 1`] = `
<DocumentFragment>
  <a
    class="link link--default-red"
    data-testid="externalLinkComponent"
    href="http://external.com"
    rel="noopener noreferrer nofollow"
    target="_blank"
  >
    External link
  </a>
</DocumentFragment>
`;

exports[`Link component renders correctly for external link with type main 1`] = `
<DocumentFragment>
  <a
    class="link link--main"
    data-testid="externalLinkComponent"
    href="http://external.com"
    rel="noopener noreferrer nofollow"
  >
    External link
  </a>
</DocumentFragment>
`;

exports[`Link component renders correctly for external link with type main that should open in a new tab 1`] = `
<DocumentFragment>
  <a
    class="link link--main"
    data-testid="externalLinkComponent"
    href="http://external.com"
    rel="noopener noreferrer nofollow"
    target="_blank"
  >
    External link
  </a>
</DocumentFragment>
`;

exports[`Link component renders correctly for internal link with type accent 1`] = `
<DocumentFragment>
  <a
    class="link link--accent"
    data-testid="internalLinkComponent"
    href="/"
  >
    Internal link
  </a>
</DocumentFragment>
`;

exports[`Link component renders correctly for internal link with type accent that should open in a new tab 1`] = `
<DocumentFragment>
  <a
    class="link link--accent"
    data-testid="internalLinkComponent"
    href="/"
    rel="noopener noreferrer"
    target="_blank"
  >
    Internal link
  </a>
</DocumentFragment>
`;

exports[`Link component renders correctly for internal link with type brand 1`] = `
<DocumentFragment>
  <a
    class="link link--brand"
    data-testid="internalLinkComponent"
    href="/"
  >
    Internal link
  </a>
</DocumentFragment>
`;

exports[`Link component renders correctly for internal link with type brand that should open in a new tab 1`] = `
<DocumentFragment>
  <a
    class="link link--brand"
    data-testid="internalLinkComponent"
    href="/"
    rel="noopener noreferrer"
    target="_blank"
  >
    Internal link
  </a>
</DocumentFragment>
`;

exports[`Link component renders correctly for internal link with type default 1`] = `
<DocumentFragment>
  <a
    class="link link--default"
    data-testid="internalLinkComponent"
    href="/"
  >
    Internal link
  </a>
</DocumentFragment>
`;

exports[`Link component renders correctly for internal link with type default that should open in a new tab 1`] = `
<DocumentFragment>
  <a
    class="link link--default"
    data-testid="internalLinkComponent"
    href="/"
    rel="noopener noreferrer"
    target="_blank"
  >
    Internal link
  </a>
</DocumentFragment>
`;

exports[`Link component renders correctly for internal link with type default-red 1`] = `
<DocumentFragment>
  <a
    class="link link--default-red"
    data-testid="internalLinkComponent"
    href="/"
  >
    Internal link
  </a>
</DocumentFragment>
`;

exports[`Link component renders correctly for internal link with type default-red that should open in a new tab 1`] = `
<DocumentFragment>
  <a
    class="link link--default-red"
    data-testid="internalLinkComponent"
    href="/"
    rel="noopener noreferrer"
    target="_blank"
  >
    Internal link
  </a>
</DocumentFragment>
`;

exports[`Link component renders correctly for internal link with type main 1`] = `
<DocumentFragment>
  <a
    class="link link--main"
    data-testid="internalLinkComponent"
    href="/"
  >
    Internal link
  </a>
</DocumentFragment>
`;

exports[`Link component renders correctly for internal link with type main that should open in a new tab 1`] = `
<DocumentFragment>
  <a
    class="link link--main"
    data-testid="internalLinkComponent"
    href="/"
    rel="noopener noreferrer"
    target="_blank"
  >
    Internal link
  </a>
</DocumentFragment>
`;
